lib/vector-buffers/src/variants/disk_v1/tests/mod.rs

```diff
@@ -113,3 +113,88 @@ macro_rules! assert_reader_v1_delete_position {
         );
     }};
 }
+
+#[macro_export]
+macro_rules! assert_buffer_usage_metrics {
+    ($usage:expr, @asserts ($($field:ident => $expected:expr),*) $(,)?) => {{
+        let usage_snapshot = $usage.snapshot();
+        $(
+            assert_eq!($expected, usage_snapshot.$field);
+        )*
+    }};
+    ($usage:expr, @asserts ($($field:ident => $expected:expr),*), recv_events => $recv_events:expr, $($tail:tt)*) => {{
+        assert_buffer_usage_metrics!(
+            $usage,
+            @asserts ($($field => $expected,)* received_event_count => $recv_events),
+            $($tail)*
+        );
+    }};
+    ($usage:expr, @asserts ($($field:ident => $expected:expr),*), recv_events => $recv_events:expr) => {{
+        assert_buffer_usage_metrics!(
+            $usage,
+            @asserts ($($field => $expected,)* received_event_count => $recv_events)
+        );
+    }};
```

Comment:
Can these (and other pairs below) be combined by eliminating the `,` and adding an empty base case?
```suggestion
    ($usage:expr, @asserts ($($field:ident => $expected:expr),*), recv_events => $recv_events:expr $($tail:tt)*) => {{
        assert_buffer_usage_metrics!(
            $usage,
            @asserts ($($field => $expected,)* received_event_count => $recv_events)
            $($tail)*
        );
    }};
```

Reply:
It's not possible to use that specific syntax -- expressions need terminating tokens when the macro definition also allows arbitrary tokens (TTs) to come after the expression -- but I can just slim it down so that all field/expectation pairs have to be terminated with a comma, since that's trivial and not really a burden at all.

Reply:
Addressed in 6da55f3cf.